lib/dockerregistry/inventory.go

```diff
@@ -106,6 +106,23 @@ func MakeSyncContext(
 	return sc, nil
 }
 
+// CollectErrors accepts any number of Errors as input which are then
+// stored in the SyncContext's Logs field.
+func (sc *SyncContext) CollectErrors(errors ...Error) {
+	for _, err := range errors {
+		sc.Logs.Errors = append(sc.Logs.Errors, err)
+	}
+}
+
+// LogJSONSummary logs the SyncContext's Logs as a prettified JSON.
+func (sc *SyncContext) LogJSONSummary() {
+	json, err := json.MarshalIndent(sc.Logs, "", "\t")
+	if err != nil {
+		klog.Info("There was a problem generating the JSON summary.")
```

Comment:
Always always always log the error as well. It is very bad practice to hide information from errors, unless of course such errors contain sensitive information (which is not the case here).